### PR TITLE
fix(ingestion/redshift): fix schema field data type mappings

### PR DESCRIPTION
**Pull request by siddiquebagwan-gslab**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py [hunk 1]
```diff
@@ -641,6 +644,7 @@ def gen_schema_fields(self, columns: List[RedshiftColumn]) -> List[SchemaField]:
             if col.sort_key:
                 tags.append(TagAssociationClass(make_tag_urn(Constants.TAG_SORT_KEY)))
 
+            logger.debug(f"col.data_type = {col.data_type}")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py:647]: let's not log this - will be overwhelming

**Review by hsheth2 — COMMENTED** [review]
can you also do an audit to check if there's any other redshift data types that we might be missing here?

File: metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py (hunk 1 above)
Comment (siddiquebagwan-gslab) [reply metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py]: done

**Comment by siddiquebagwan-gslab** [comment]
> can you also do an audit to check if there's any other redshift data types that we might be missing here?

 Added more types and for GEOMETRY, HLLSKETCH set the type as NullType because GEOGRAPHY is set to NullType

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**